packages/thirdweb/src/react/web/ui/components/WalletImage.tsx

```diff
@@ -15,15 +23,46 @@ export function WalletImage(props: {
   id: WalletId;
   size: string;
   client: ThirdwebClient;
+  allowOverrides?: boolean;
 }) {
-  const mipdImage = getInstalledWalletProviders().find(
-    (provider) => provider.info.rdns === props.id,
-  )?.info.icon;
+  const [image, setImage] = useState<string | undefined>(undefined);
+  useEffect(() => {
+    async function fetchImage() {
+      let mipdImage = getInstalledWalletProviders().find(
+        (provider) => provider.info.rdns === props.id,
+      )?.info.icon;
 
-  if (mipdImage) {
+      if (props.allowOverrides && props.id === "inApp") {
+        // check last auth provider and override the IAW icon
+        const lastAuthProvider = await getLastAuthProvider();
+        switch (lastAuthProvider) {
+          case "google":
+            mipdImage = googleIconUri;
+            break;
+          case "apple":
+            mipdImage = appleIconUri;
+            break;
+          case "facebook":
+            mipdImage = facebookIconUri;
+            break;
+          case "phone":
+            mipdImage = phoneIcon;
+            break;
+          case "email":
+            mipdImage = emailIcon;
+            break;
+        }
+      }
+
+      setImage(mipdImage);
+    }
+    fetchImage();
+  }, [props.id, props.allowOverrides]);
```

Comment:
We can replace state + effect to react-query here

Reply:
so i tried to do that, but the query key is tricky for this because it comes from local storage, so need to invalidate when the value gets written, bit messy.

wdyt?